.travis.yml

```diff
@@ -9,10 +10,38 @@ env:
   - secure: f1lofDZd+7PHPcV011Bq4nMS2YUcWC231bVSFIVdAcgj9K92kyGLjeyWneJmHHofpM+09R49w8WYeA7M+8xd9E8/qoSeLmYM8FZrSRRn12nz75fvfHsN+M2G4RZeC2DcYH9uUjARUj4o/+PiUHfTF+mG1Zcad3uJBAPzXrSZbwl9b5rkB4uF/8+iKTVTzMvGqwapYJonRDaWpp7jQYYSl7Tl5duE2iYFr3qa1EfBc1IBjOxTHr05I3PhtOu1c16REL08kKZrJIcpmiETinA2ETq1rQJaxnVqLM+5td9nVJ5uBkwBlTli21+MDjVH5Y5ZSofwVe/dHayFce5sgSHLXJxLBIt/fBODCXELIPG0AyUmQfDpZAAWMm9tKkbwxfJrQrfcYhoZWBi866Q0cEXYVDTD0xJpZlRDGRLYlMTLWjumq8BQXh9o0GnwGX/f5siJAQ1K4wtDledWvFGKOQkLfVfgArynbl5S91xn179VZ/OcFMOIpzPCxNehZ2IfhS16H+a+WGOiRtflil+2YbH/1qUM/ZHxeIWVWh5uBJvZ5QxutzTEG24G2JRQF9Ph1uN4a9djRTmT+PwaF6R9uvxz9PR2jsAW1Yn5c88/cz7DvtTjOHu7VDAaw8CBM/Wl81F6AmP/NqOdI2IWyfXrbTQdnWjkqpGzZtEP8quVpt9DFys=
   - secure: MvJDJRn23aWIVtcZVRg80nFYrk9wDlLAI6AZWgbCHfoVSu/PmoOnTVALxQJ1qVm2ywUsSvn0Oc08slGNfweRf4bhfeq9NFx5zxB3KNwxMPH66jU3Dri02tHXECxH9/cwyD5g8Pkwh2oVIWXbjgiNYQgFzcDLyhgToIBSCGMIkNdvAoK9lRVmKScwPQp5LqeolFbw8FcTcN9ozRsM7RewQXloo/FmUKMpBc5qRaZgEdnj16qYqCwXgQIrDmt0fixhdyvWBHWeTssO7jvcSuhCkjgbO0OinKID+299Bekm6fIoISFLptTHfJHdCRA7sBPv3vqP3VuA2h71PDReIbDg6pjymAwWACXy9Dl/dLnx5CbkGawym8ZzpLxXUBLqYcrEy6fsigFOH2BWkFAKHPEEjHS2cwHfHmoftTwhN2U4MkK8SYL5a2w0y5EjaETG20m7/SIyWq094SVhpRJhDjEb9oW61LtcH+CODNJzsHf3F8AZH1Igew+zutJlkC30fEKJ61yvS69LZmhSNCLpGAosmZWyR41t0S2XYHPb6M8VDJrezl2E99UsHrJkepfjhP9whbZHstbToh/m8xhKHb0Ple6gRYZ7uqOiPbOc9FjN7pH/sNDn3n9fymS8qXddJK2qd/rAjJ9St+4mzq/5heEgP/gyjQwwOcFul32jC7ds7Tg=
   - secure: V0zGqKSrp6faxVVQd8nW16+5XgeiyC1ZXojkCN4V91GDh7FhUfkM63liEpQhGOnRnQ9Z+ibmCSiTvS8kSCjkuME7rKC7jfmvEnsIQhGGk7GALoZfjP3Ox78czGFkZ8C3e6Y6rBStVoZDXXj6QpxWQxvYbFsf295g8QXjOliRfUchGz/H++OMbOcM1vNWQCfGNsRQ+0xWDYbk/z2V1XVGU1yxuCPDozWiwH2JjU4i8zzAhaqFua7ikfmneDhrTqd9BpR5cTfww66Tvb8KfqmQRrXoNelYMGoOqz/raVxP2+WtJr1sIhT/q6/rX9Bizl6+S5Pn7h9YES5eNOmIkIHwDgiP4SUwnT4PwuJw2PyAVCzn0WHceLGevHs8Wid82ONlXfON5gNJSpVUzcy7cym0LPbA60wL0MxjICmWXJklgdYqIruYZX9ijscwH4rWia4Q83n3BcSyzHr4g37Fu0f0W5cqcinNRV+dtQ0LdZA9yFH2qDMV1yngrUmDcnlMvMKnRwzDITzauDM6TmAXfphkNOTR32c1g4Rl84EFAqGtYI6tQoAAUm3ZEEMJzcL8gviCWqzQx3UkRq4UP2C7yF21p/2RqrhdAkeZU55J6OhymCdFxEjJswNlUj0y1u+mSw0ARmAO4ZtZ7XCQ1G3sBcJz8D4d4gx1Hc/HzHPERn8oZpE=
-after_success:
-- source ./build/travis-utils.sh
-- if is_travis_push_env; then docker login -u $DOCKER_USER -p $DOCKER_PASSWORD; fi
-- if is_travis_push_env; then make image-quick; fi
-- if is_travis_push_env; then make push; fi
-- if is_travis_push_env; then make tag-latest; fi
-- if is_travis_push_env; then make push-latest; fi
+jobs:
+  include:
+    - stage: "build, test, deploy"
+      script: make
+      deploy:
+        - provider: script
+          skip_cleanup: true
+          script: make deploy-travis
+          on:
+            repo: open-policy-agent/opa-istio-plugin
+            branch: master
+            tags: false
+    - stage: e2e
+      if: (NOT type IN (pull_request)) AND (branch = master)
+      before_script:
```

Comment:
Is it possible to put this stuff in `before_script` as a make target and/or shell script in the repo? Seems like it would be pretty useful to have a like one-liner that sets up a kind cluster with istio (with the correct versions) for local dev/test of the plugin.

Reply:
Created a make target `test-cluster` which creates a Kubernetes cluster using kind and deploys Istio on top of it. But when I included this target as part of Travis's `before_script`, the Travis `script` (ie. `make test-e2e`) fails and is unable to connect to the Kube cluster. 

Same result is seen if `make test-cluster` is made part of the Travis `script`. So I'm going to keep the `before_script` as is for now and we can re-visit this later.  Also keeping the new make target `test-cluster` for local testing.